### ISSUE TITLE
Trig functions on uncertainty.Distribution objects give error

### ISSUE THREAD

**Issue by rikvl** [astropy/astropy#12236]
### Description

Passing an `astropy.uncertainty.Distribution` object to the `numpy` trigonometric functions `np.sin`, `np.cos`, `np.tan` gives an error in astropy v4.3, while these operations work fine in v4.2

### Steps to Reproduce

```python
import numpy as np

from astropy import units as u
from astropy import uncertainty as unc

angles = unc.Distribution([90., 30., 0.]*u.deg)

sin_angles = np.sin(angles)

print(sin_angles)
```

### Expected behavior

In `v4.2.x` the above code produces the output
```
[1.  0.5 0. ] with n_samples=3
```

### Actual behavior

In `v4.3.x` this is the error message:
```
Traceback (most recent call last):
  File "/Users/rik/cita/astropy/../test-np-unc.py", line 8, in <module>
    sin_angles = np.sin(angles)
  File "/Users/rik/cita/astropy/astropy/units/quantity.py", line 483, in __array_ufunc__
    arrays.append(converter(input_) if converter else input_)
  File "/Users/rik/cita/astropy/astropy/units/core.py", line 1045, in <lambda>
    return lambda val: scale * _condition_arg(val)
  File "/Users/rik/cita/astropy/astropy/units/core.py", line 2552, in _condition_arg
    raise ValueError("Value not scalar compatible or convertible to "
ValueError: Value not scalar compatible or convertible to an int, float, or complex array
```

The problem seems to arise from [changes](https://github.com/astropy/astropy/pull/11240/files#diff-e64d6a061c789fa95df831a50cd3d420a07ac6a9831c060e60372aded1ebdeeb) made on the function
`get_converter` in `astropy/units/quantity_helper/helpers.py` in PR #11240

### System Details

macOS-10.14.6-x86_64-i386-64bit
Python 3.9.1 (default, Jan 28 2021, 20:35:40)
[Clang 10.0.1 (clang-1001.0.46.4)]
Numpy 1.19.5
astropy 4.3.1
Scipy 1.6.0
Matplotlib 3.4.1

**Comment by mhvk** [comment]
Interesting, the problem here is that the Distribution is a scalar; it works for `angles = unc.Distribution([[90., 30., 0.]]*u.deg)`.

**Comment by mhvk** [comment]
Fix in #12471
